engine/crates/engine-v2/schema/src/builder/mod.rs

```diff
@@ -502,16 +489,17 @@ impl From<federated_graph::Value> for Value {
         match value {
             federated_graph::Value::String(s) => Value::String(s.into()),
             federated_graph::Value::Int(i) => Value::Int(i),
-            federated_graph::Value::Float(f) => Value::Float(f.into()),
+            federated_graph::Value::Float(f) => Value::Float(f),
             federated_graph::Value::Boolean(b) => Value::Boolean(b),
             federated_graph::Value::EnumValue(s) => Value::EnumValue(s.into()),
             federated_graph::Value::Object(fields) => Value::Object(
                 fields
+                    .into_vec()
                     .into_iter()
                     .map(|(id, value)| (id.into(), value.into()))
                     .collect(),
             ),
-            federated_graph::Value::List(l) => Value::List(l.into_iter().map(Into::into).collect()),
+            federated_graph::Value::List(l) => Value::List(l.into_vec().into_iter().map(Into::into).collect()),
```

Comment:
why `into_vec()`?

Reply:
`Box::<[T]>::into_iter()` gives you an iterator over `&T`s, but we were expecting owned iterators in these spots. `into_vec()` does that without copy / allocation. Happy to change if there's a better option though.